improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py

```diff
@@ -255,6 +259,78 @@ def test_insufficient_forecast_count(self):
         self.assertIsNone(result[1])
 
 
+class Test__interpolate(unittest.TestCase):
+
+    """Test the _interpolate method."""
+
+    def setUp(self):
+
+        """Set up data for testing the interpolate method."""
+
+        self.reliability_probabilities = np.array([0.0, 0.4, 0.8])
+        self.observation_frequencies = np.array([0.2, 0.6, 1.0])
+        self.plugin = Plugin()
+
+    def test_unmasked_data(self):
+        """Test unmasked data is interpolated and returned as expected."""
+
+        expected = np.array([0.4, 0.6, 0.8])
+        forecast_threshold = np.array([0.2, 0.4, 0.6])
+
+        result = self.plugin._interpolate(forecast_threshold,
+                                          self.reliability_probabilities,
+                                          self.observation_frequencies)
+
+        assert_allclose(result, expected)
+
+    def test_masked_data(self):
+        """Test masked data is interpolated and returned with the original
+        mask in place."""
+
+        expected = np.ma.masked_array([0.4, 0.6, 0.8], mask=[1, 0, 0])
+        forecast_threshold = np.ma.masked_array([0.2, 0.4, 0.6],
+                                                mask=[1, 0, 0])
+
+        result = self.plugin._interpolate(forecast_threshold,
+                                          self.reliability_probabilities,
+                                          self.observation_frequencies)
+
+        assert_allclose(result, expected)
+        assert_allclose(result.data, expected.data)
```

Comment:
Do you need to check result.data.data and result.data.mask separately?

Reply:
Not really, but in this case I was trying to demonstrate that the interpolation is still done with whatever point is underneath the mask (valid or NaN), hence the importance of putting the mask back. I'll make it a NaN to make the point a bit more obvious (plus who knows what it will actually throw back at us; let's find out).